frontend/src/metabase/common/components/FilterPicker/FilterPicker.unit.spec.tsx

```diff
@@ -227,4 +242,53 @@ describe("FilterPicker", () => {
       );
     });
   });
+
+  describe("custom expression", () => {
+    async function editExpressionAndSubmit(text: string) {
+      const input = screen.getByLabelText("Expression");
+      const button = screen.getByRole("button", { name: "Done" });
+
+      // The expression editor applies changes on blur,
+      // but for some reason it doesn't work without `act`.
+      // eslint-disable-next-line testing-library/no-unnecessary-act
+      await act(async () => {
+        await userEvent.type(input, text);
+        await userEvent.tab();
+      });
+
+      await waitFor(() => expect(button).toBeEnabled());
+      userEvent.click(button);
+    }
```

Comment:
This drives me crazy — I've tried many different ways to make this work, but it seems like only a combination of `act` with `userEvent.tab` and `waitFor` did the thing :/

TLDR: the expression editor's "Done" button becomes enabled once the expression is valid. The expression is submitted for validation on the input's blur event. So we should edit > blur > click "Done"

What I've tried:

- edit > click "Done"
- edit > `fireEvent.blur(input) > click "Done"
- edit > `userEvent.tab()` > click "Done"
- edit > `fireEvent.focus(input) > `fireEvent.blur(input)` > click "Done"

Neither of them worked. In all cases, the "Done" button remained disabled, even with:

```js
await waitFor(() => expect(button).toBeEnabled());
```

The only two things that worked are `act` or something like that:

```jsx
render(
  <>
     <FilterPicker {...props} />
     <input data-testid="__TEST_ONLY_HACK_INPUT" />
  </>
)

userEvent.type(expressionInput, formula)
userEvent.type(screen.getByTestId("__TEST_ONLY_HACK_INPUT", "")
userEvent.click(doneButton)
```